src/test/java/de/tum/in/www1/artemis/ModelingExerciseIntegrationTest.java

```diff
@@ -383,4 +391,37 @@ public void createModelingExercise_setNeitherCourseAndExerciseGroup_badRequest()
         ModelingExercise modelingExercise = ModelFactory.generateModelingExerciseForExam(DiagramType.ClassDiagram, null);
         request.postWithResponseBody("/api/modeling-exercises/", modelingExercise, ModelingExercise.class, HttpStatus.BAD_REQUEST);
     }
+
+    @Test
+    @WithMockUser(value = "instructor2", roles = "INSTRUCTOR")
+    public void searchModelingExercises_instructor_shouldOnlyGetResultsFromOwningCourses() throws Exception {
+        final var result = configureSearchAndReturnResult("");
+        assertThat(result.getResultsOnPage()).isEmpty();
+    }
+
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void searchModelingExercises_instructor_getResultsFromOwningCoursesNotEmpty() throws Exception {
+        final var result = configureSearchAndReturnResult("ClassDiagram");
+        assertThat(result.getResultsOnPage().size()).isEqualTo(1);
+    }
+
+    @SuppressWarnings("unchecked")
+    private SearchResultPageDTO<ModelingExercise> configureSearchAndReturnResult(String searchTerm) throws Exception {
+        final var search = new PageableSearchDTO<String>();
```

Comment:
This is a lot of duplicated code. Please create a helper method and avoid duplicated code!